### ISSUE TITLE
Better error codes on failure

### ISSUE THREAD

**Issue by gbiellem** [SharpTools/SharpReverseProxy#16]
@andrecarlucci Inside the middleware [here](https://github.com/SharpTools/SharpReverseProxy/blob/master/src/SharpReverseProxy/ProxyMiddleware.cs#L50) there is a catch all on any failure with the proxyrequest. It returns a HTTP Response code of 503.  

Would it be better to inspect the exception and return a more meaningful error.
Something like this maybe?

```
            try
            {
                await ProxyTheRequest(context, proxyRequest, matchedRule).ConfigureAwait(false);
            }
            catch (HttpRequestException ex) when  ((ex.InnerException as WebException)?.Status == WebExceptionStatus.ConnectFailure)
            {
                //Upstream server is down
                context.Response.StatusCode = StatusCodes.Status504GatewayTimeout;
            }
            catch(HttpRequestException)
            {
                //Catch all
                context.Response.StatusCode = StatusCodes.Status503InternalServerError;
            }
            _options.Reporter.Invoke(resultBuilder.Proxied(proxyRequest.RequestUri, context.Response.StatusCode));



```

**Comment by andrecarlucci** [comment]
Hi @gbiellem, 

I like the idea. 
Maybe also record the reason phrase and pass the original exception to the reporter.

Are you using the code you provided? Is it working well?

**Comment by andrecarlucci** [comment]
Solved on v2.0
